runtime/src/snapshot_utils.rs

```diff
@@ -1041,13 +1041,14 @@ fn deserialize_snapshot_data_files_capped<T: Sized>(
                     incremental_snapshot_root_file_path,
                     maximum_file_size,
                 )?;
-            (
-                Some(incremental_snapshot_file_size),
-                Some(incremental_snapshot_data_file_stream),
-            )
+            Some((
```

Comment:
we should just return directly, something like:
`Some(create_snpashot_data_file_stream(...)?)`

I think because of the `?` we cannot easily use map here, but mapping would also let us get rid of the `else { None }`